REVIEW_CHECKLIST.md

```diff
@@ -1,3 +0,0 @@
-# Review Checklist
-
-Refer to the [Collection review checklist](https://github.com/ansible/community-docs/blob/main/review_checklist.rst).
```

Comment:
@gotmax23 thanks for the PR, how about putting a link to docs ansible com's page where the checklist currently lives? Maybe not in this doc but in README for example? Reviewing others works is a very important contribution we have a lack of:)

Reply:
@gotmax23 it's review checklist for issues/PRs, not for collection inclusion.
The file used to contain the review checklist itself but then the checklist was moved to another repo and then to docs.ansible.com. 
The link should be there, somewhere.
You could:
1) update the link in the file to point to [docs.ansible.com](https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_reviewing.html) OR
2) put the link in README and remove the file, though I personally like having it in a separate file

Reply:
I combined the REVIEW_CHECKLIST.md and CONTRIBUTING.md, fixed the links, and clarified what the review checklist is actually for. I was confused by the previous text. I combined them, because both files were pretty short, and these are related. I can split it back out if you'd like.